lib/src/http.dart

```diff
@@ -37,6 +44,20 @@ class _PubHttpClient extends http.BaseClient {
       : this._inner = inner == null ? new http.Client() : inner;
 
   Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    if (_shouldAddMetadata(request)) {
+      request.headers["X-Pub-OS"] = Platform.operatingSystem;
+      request.headers["X-Pub-Command"] = PubCommandRunner.command.join(" ");
+      request.headers["X-Pub-Session-ID"] = _sessionId;
+
+      var environment = Platform.environment["PUB_ENVIRONMENT"];
+      if (environment != null) {
+        request.headers["X-Pub-Environment"] = environment;
```

Comment:
> Generally speaking I think we should sanitize the data that goes into the headers (e.g. could this environment variable have PII data in it?) and also what format we use: To avoid any issues we should e.g. restrict ourselves to ASCII-only header values.

The environment variable is scoped to this application, so I don't think it's likely anyone's already using it for anything, much less for storing PII.

I think allowing non-ASCII characters here is fine. They aren't guaranteed to be well-preserved through proxies, but they won't break the request and we don't expect to need to track them effectively.